hostpool.go

```diff
@@ -171,6 +174,19 @@ func (p *standardHostPool) SetHosts(hosts []string) {
 	p.setHosts(hosts)
 }
 
+func (p *standardHostPool) SetErrorBudget(maxFailures int, failureWindow time.Duration) {
+	p.Lock()
+	defer p.Unlock()
+
+	p.maxFailures = maxFailures
+	p.failureWindow = failureWindow
+
+	// hosts and hostList are pointers to the same hostEntry so we only need update one.
+	for _, h := range p.hostList {
+		h.failures = NewRingBuffer(p.maxFailures)
```

Comment:
Yeah, fair comment. I was trying to avoid changing the pool initialisation code at this point, but since we want to add config options it makes sense to move combine it with this 👍